source/common/network/io_socket_error_impl.cc

```diff
@@ -8,7 +8,34 @@
 namespace Envoy {
 namespace Network {
 
-Api::IoError::IoErrorCode IoSocketError::getErrorCode() const {
+Api::IoError::IoErrorCode IoSocketError::getErrorCode() const { return error_code_; }
+
+std::string IoSocketError::getErrorDetails() const { return errorDetails(errno_); }
+
+IoSocketError* IoSocketError::getIoSocketEagainInstance() {
+  static auto* instance = new IoSocketError(SOCKET_ERROR_AGAIN, Api::IoError::IoErrorCode::Again);
+  return instance;
+}
+
+void IoSocketError::deleteIoError(Api::IoError* err) {
+  ASSERT(err != nullptr);
+  if (err != getIoSocketEagainInstance()) {
+    delete err;
+  }
+}
+
+inline IoSocketError* getIoSocketInvalidAddressInstance() {
+  static auto* instance =
+      new IoSocketError(SOCKET_ERROR_NOT_SUP, Api::IoError::IoErrorCode::NoSupport);
+  return instance;
+}
+
+Api::IoCallUint64Result IoSocketError::ioResultSocketInvalidAddress() {
+  return Api::IoCallUint64Result(
+      0, Api::IoErrorPtr(getIoSocketInvalidAddressInstance(), [](IoError*) {}));
+}
+
+Api::IoError::IoErrorCode IoSocketError::errorCodeFromErrno() const {
```

Comment:
`ECONNRESET`/`WSAECONNRESET` while debugging some integration tests. I could add them as well

Reply:
I would expect other users to notice similar log spam.  I think that in its current form this PR reduces but doesn't eliminate the log spam hence the potential value in handling those additional error codes.

Reply:
If we add this error in the list of known errors then we only log it when the level is trace.  Because `raw_buffer_socket` logs error codes only in trace level. (`source/common/network/raw_buffer_socket.cc:38`)

I don't have much experience with the log verbosity so I don't have a strong opinion here but it remote disconnects might be useful to have on debug.

Reply:
The debug log below is a plain MISC log instead of a CONN log, so at debug you would know about a ECONNRESET but not what triggered it.

I think it's fine to avoid the extra debug logs.

Reply:
Added this error in the list in febdfbc88c7bdf0f2ef1e6baf4156d5e89610bb1. That way we avoid the extra logging